### PR TITLE
Include Data Integrity test for  nvme-of targets

### PR DESCRIPTION
**Pull request by preethisn**
# Description
This does Data Integrity test on nvme-of targets luns. 
Test case Flow-
Configure subsystems
Configure Initiator
Discover subsystems
connect to subsystems
Lun discovery
Create filesystem and mount
Run FIO write on fileystsem
perform Md5sum checksum 
unmount the device
Mount it back 
Repeat for FIO read on to the same File
Capture md5sum and compare

Logs attached-
Test case log- http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-19ISZ2/Test_to_perform_Data_Integrity_test_over_NVMe-OF_targets_0.log   --Refer this
Function suit tests logs-http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-19ISZ2

### CODE REVIEW

File: tests/nvmeof/test_ceph_nvmeof_data_integrity.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+                    f"--filename={mount_point}/test.txt"
+                )
+                rbd_obj.exec_cmd(cmd=cmd_write)
+                # Calculate md5sum after FIO write
+                md5sum_write = rbd_obj.exec_cmd(cmd=f"md5sum {mount_point}/test.txt")
+                # Run FIO for reading the files
+                LOG.info("Running FIO for read operations.")
+                cmd_read = (
+                    f"fio --name=readiops --ioengine=sync "
+                    f"--rw=randwrite --refill_buffers=1 "
```
[the remark this answers is not in the document]
Comment (preethisn) [reply tests/nvmeof/test_ceph_nvmeof_data_integrity.py]: addressed

**Comment by preethisn** [comment]
latest log attached after changes made - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-U97M47

**Comment by preethisn** [comment]
I was able to run the full test suit and below logs
All test logs located here: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-N7SHOP

File: tests/nvmeof/test_ceph_nvmeof_data_integrity.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                LOG.info("Running FIO for write operations.")
+                cmd_write = f"{fio_command} --name=writeiops --rw=randwrite"
+                rbd_obj.exec_cmd(cmd=cmd_write)
+                # Calculate md5sum after FIO write
+                md5sum_write = rbd_obj.exec_cmd(cmd=f"md5sum {mount_point}/test.txt")
+                # Run FIO for reading the files
+                LOG.info("Running FIO for read operations.")
+                cmd_read = f"{fio_command} --name=readiops --rw=randread"
+                rbd_obj.exec_cmd(cmd=cmd_read)
+                # Calculate md5sum after FIO read
```
Comment (Manohar-Murthy) [inline tests/nvmeof/test_ceph_nvmeof_data_integrity.py:217]: Can we have some operations done before comparing the md5sum? 
Ex: Network restarts,service restarts,data unavilability (unmounts) scenarios.

Comment (preethisn) [reply tests/nvmeof/test_ceph_nvmeof_data_integrity.py:222]: Yes added unmounts before comparing - Will take other scenarios as separate test case

**Comment by preethisn** [comment]
latest logs after unmount change is added  -http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-0KEBZO

**Review by Manohar-Murthy — APPROVED** [review]
lgtm.

**Pull request merged**